hack/pre-run-e2e.sh

```diff
@@ -0,0 +1,266 @@
+#!/usr/bin/env bash
+
+# Copyright 2022 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+if [[ "${REGISTRY_USER_NAME}" = "" ]] || [[ "${REGISTRY_PASSWORD}" = "" ]]; then
+  echo "REGISTRY_USER_NAME and REGISTRY_PASSWORD can not be empty, please set it"
+  exit 1
+fi
+
+export REGISTRY_SERVER_ADDRESS=swr.ap-southeast-1.myhuaweicloud.com
+export REGISTRY=swr.ap-southeast-1.myhuaweicloud.com/dev-registry
+export VERSION=latest
+
+echo -e "\nBuild images"
+make upload-images
+
+echo -e "\nCheck cloud-config secret"
+count=`kubectl get -n kube-system secret cloud-config | grep cloud-config | wc -l`
+if [[ "$count" -ne 1 ]]; then
+  echo "Please create a secret with the name: cloud-config."
+  exit 1
+fi
+
+kubectl delete -n kube-system daemonset --ignore-not-found=true huawei-cloud-controller-manager
```

Comment:
Sorry, I should explain it.
The provider usually runs on the master nodes. Using the `Daemonset` with selectors is an easy way. 
Using `Deployment` is also possible, but if we scale, there will be no nodes available to create new pods.

Reply:
Yeah, I agree the provider usually runs on master nodes. But not required

> Using the Daemonset with selectors is an easy way.

I guess you mean `nodeSelector`:
```yaml
      nodeSelector:
        node-role.kubernetes.io/master: ""
```
`Deployment` can do that as well.

I still think `Daemonset` is not a good idea. Since the semantic meaning of `Daemonset` is run `on every node`.

Reply:
OK, get it, thanks.

Reply:
I have changed to `Deployment` way.